docs/asciidoc-writers-guide.adoc

```diff
@@ -366,6 +366,8 @@ include::{includedir}/xref.adoc[]
 
 include::{includedir}/xref-relative.adoc[]
 
+TIP: The xref inline macro by default resolves the link to HTML, to create a relative link to an asciidoc file in another subdirectory include `ifdef::env-github[:outfilesuffix: .adoc]` below the document header and drop the ".adoc" extension after the filename.
```

Comment:
It's important to note that the xref macro only links to another file if the target ends in a hash path (e.g., `#` or `#anchor-name`). The `#` is how Asciidoctor distinguishes between an internal xref and an external xref.

> Is there a method which is more neutral (support any output or source format).

The `env-*` attributes are set by the integrations. For example, the code that runs on GitHub (and GitLab) sets the `env-github` attribute. You should expect that each integration has a similar attribute. For example, Awestruct sets `env-site`. We've tried to introduce generic attributes where possible, but in some cases like GitHub, the environment is so unique that it really only applies to that specific integration.

Btw, the browser previews also need the `outfilesuffix` to be `.adoc`. Both the Chrome and Firefox plugin set this value by default now. You can also use `env-browser` in those environments.